### PR TITLE
Feat/node test runner

### PR DESCRIPTION
**Pull request by ilteoood**
Migrate every test to node test runner, removing tap.

Additionally, tests that were relying to `http.get` have been converted to use `fetch`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,9 +10,7 @@
     "bench:cmp:ci": "node ./benchmark/compare-branches.js --ci",
     "test:lint": "standard",
     "test:typescript": "tsd",
-    "test": "standard && tap --allow-incomplete-coverage test/*.test.js && npm run test:typescript",
-    "test:report": "tap -J test/*.test.js --cov --coverage-report=html --coverage-report=cobertura | tee out.tap",
-    "test:reporter": "tap-mocha-reporter xunit < out.tap > test/junit-testresults.xml"
```
Comment (ilteoood) [inline package.json:15]: These 2 commands have been removed as they are not working anymore, and no one is using them in the CI

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
thx!

**Comment by mcollina** [comment]
thx!

**Pull request merged**